devito/interfaces.py

```diff
@@ -17,7 +18,34 @@
 _SymbolCache = {}
 
 
-class SymbolicData(Function):
+class CachedSymbol(object):
+    """Base class for symbolic objects that caches on the class type."""
+
+    @classmethod
```

Comment:
Probably a silly question

I'm not sure I understand why are the methods below class methods ? It seems like that they are used on instances later on (e.g., in SymbolicData)